src/test/java/org/cyclonedx/parsers/XmlParserTest.java

```diff
@@ -69,6 +69,15 @@ public void testValid12Bom() throws Exception {
         Assert.assertTrue(valid);
     }
 
+    @Test
+    public void testValid12BomWithPedigree() throws Exception {
+        final File file = new File(this.getClass().getResource("/bom-1.2-pedigree.xml").getFile());
+        final XmlParser parser = new XmlParser();
+        final boolean valid = parser.isValid(file, CycloneDxSchema.Version.VERSION_12);
+        final Bom bom = parser.parse(file);
```

Comment:
Maybe move the assertion on `valid` up, and add some assertions on the `bom` at the end. e.g. (ignoring `import sort madness`)

```patch
Index: src/test/resources/bom-1.2-pedigree.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/test/resources/bom-1.2-pedigree.xml b/src/test/resources/bom-1.2-pedigree.xml
--- a/src/test/resources/bom-1.2-pedigree.xml	(revision a2fe9c4287ed6c77a602c804978c506d4b490082)
+++ b/src/test/resources/bom-1.2-pedigree.xml	(date 1614192018178)
@@ -11,21 +11,21 @@
         <ancestors>
           <component type="library">
             <group>org.example</group>
-            <name>sample-library</name>
+            <name>sample-library-ancestor</name>
             <version>1.0.0</version>
           </component>
         </ancestors>
         <descendants>
           <component type="library">
             <group>org.example</group>
-            <name>sample-library</name>
+            <name>sample-library-descendant</name>
             <version>1.0.0</version>
           </component>
         </descendants>
         <variants>
           <component type="library">
             <group>org.example</group>
-            <name>sample-library</name>
+            <name>sample-library-variant</name>
             <version>1.0.0</version>
           </component>
         </variants>
Index: src/test/java/org/cyclonedx/parsers/XmlParserTest.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/test/java/org/cyclonedx/parsers/XmlParserTest.java b/src/test/java/org/cyclonedx/parsers/XmlParserTest.java
--- a/src/test/java/org/cyclonedx/parsers/XmlParserTest.java	(revision a2fe9c4287ed6c77a602c804978c506d4b490082)
+++ b/src/test/java/org/cyclonedx/parsers/XmlParserTest.java	(date 1614192018182)
@@ -18,13 +18,21 @@
  */
 package org.cyclonedx.parsers;
 
+import java.io.File;
+import java.util.List;
+
 import org.apache.commons.io.IOUtils;
 import org.cyclonedx.CycloneDxSchema;
-import org.cyclonedx.model.*;
+import org.cyclonedx.model.Bom;
+import org.cyclonedx.model.Component;
+import org.cyclonedx.model.Dependency;
+import org.cyclonedx.model.ExternalReference;
+import org.cyclonedx.model.OrganizationalContact;
+import org.cyclonedx.model.OrganizationalEntity;
+import org.cyclonedx.model.Service;
+import org.cyclonedx.model.ServiceData;
 import org.junit.Assert;
 import org.junit.Test;
-import java.io.File;
-import java.util.List;
 
 @SuppressWarnings("deprecation")
 public class XmlParserTest {
@@ -74,8 +82,11 @@
         final File file = new File(this.getClass().getResource("/bom-1.2-pedigree.xml").getFile());
         final XmlParser parser = new XmlParser();
         final boolean valid = parser.isValid(file, CycloneDxSchema.Version.VERSION_12);
+        Assert.assertTrue(valid);
         final Bom bom = parser.parse(file);
-        Assert.assertTrue(valid);
+        Assert.assertEquals("sample-library-ancestor", bom.getComponents().get(0).getPedigree().getAncestors().getComponents().get(0).getName());
+        Assert.assertEquals("sample-library-descendant", bom.getComponents().get(0).getPedigree().getDescendants().getComponents().get(0).getName());
+        Assert.assertEquals("sample-library-variant", bom.getComponents().get(0).getPedigree().getVariants().getComponents().get(0).getName());
     }
 
     @Test
```

Reply:
In 101e8d8 and a445500